src/ocean/io/serialize/StructSerializer.d

```diff
@@ -554,8 +554,9 @@ struct StructSerializer ( bool AllowUnions = false )
             {
                 // slice array (passing a static array as ref is not allowed)
                 U[] array = field;
+                alias StripAllArrays!(U) Element;
 
-                static if ( is(StripAllArrays!(U) == struct) )
+                static if ( is(Element == struct) && !isTypedef!(Element) )
```

Comment:
I'm not quite clear why we're excluding arrays of typedefs here now. Maybe a comment would help.